code/modules/mob/living/carbon/human/ai/brain/ai_brain.dm

```diff
@@ -99,41 +119,40 @@ GLOBAL_LIST_EMPTY(human_ai_brains)
 	nade_throwback(things_around)
 	item_search(things_around)
 
-	if(!currently_busy && should_reload_primary())
-		currently_busy = TRUE
+	if(!currently_reloading && should_reload_primary())
 		reload_primary()
 
 	if(primary_weapon && current_target)
-		if(allowed_approach_retreat && !has_ongoing_action(AI_ACTION_APPROACH) && !has_ongoing_action(AI_ACTION_RETREAT))
+		if(allowed_approach_retreat && !length(ongoing_actions))
 			var/target_futile = current_target.is_mob_incapacitated()
 			var/distance = get_dist(tied_human, current_target)
 			if(distance > gun_data.optimal_range || target_futile)
-				if(!in_cover)
+				if(!current_cover)
 					var/walk_distance = target_futile ? gun_data.minimum_range : gun_data.optimal_range
 					ADD_ONGOING_ACTION(src, AI_ACTION_APPROACH, current_target, walk_distance)
 
 			else if(distance < gun_data.optimal_range)
-				var/walk_distance = in_cover ? gun_data.minimum_range : gun_data.optimal_range
+				var/walk_distance = current_cover ? gun_data.minimum_range : gun_data.optimal_range
 				ADD_ONGOING_ACTION(src, AI_ACTION_RETREAT, current_target, walk_distance)
 
-		if(!currently_busy && !currently_firing && COOLDOWN_FINISHED(src, fire_overload_cooldown))
+		if(!currently_busy && !currently_firing && !currently_reloading && COOLDOWN_FINISHED(src, fire_overload_cooldown))
 			currently_busy = TRUE
 			attack_target()
 
-	if(!currently_busy && !in_combat && healing_start_check())
-		currently_busy = TRUE
-		start_healing()
-
 	if(!currently_busy && !current_target && primary_weapon)
 		current_target = get_target(view_distance)
-		RegisterSignal(current_target, COMSIG_PARENT_QDELETING, PROC_REF(on_target_delete))
+		RegisterSignal(current_target, COMSIG_PARENT_QDELETING, PROC_REF(on_target_delete), TRUE)
+
+	if(!currently_busy && !in_combat && healing_start_check())
```

Comment:
why move this

Reply:
By potentially getting a new target before starting the healing sequence - we will have more understanding of our current surrondings, which can help in case we are going to make some new complex actions/orders/behaviors/etc regarding use of closest target while treating yourself with drugs.
Conclusion: Might help in future, currently reduntant